src/sig_stfl/lms/sig_stfl_lms_functions.c

```diff
@@ -0,0 +1,813 @@
+		return OQS_ERROR;
+	}
+
+	/* Lock secret key to ensure data integrity use */
+	if ((secret_key->lock_key) && (secret_key->mutex)) {
+		secret_key->lock_key(secret_key->mutex);
+	}
+
+	oqs_key_data = secret_key->secret_key_data;
+	if (oqs_key_data == NULL) {
+		goto err;
+	}
+	priv_key = oqs_key_data->sec_key;
+	if (priv_key == NULL) {
+		goto err;
+	}
+
+	current_count = get_bigendian(priv_key + PRIVATE_KEY_INDEX, PRIVATE_KEY_INDEX_LEN /*0, 8 */);
+	*remain =  (total_sigs - (unsigned long long)current_count);
+
+err:
+	/* Unlock secret key */
+	if ((secret_key->unlock_key) && (secret_key->mutex)) {
+		secret_key->unlock_key(secret_key->mutex);
+	}
+	return OQS_SUCCESS;
+}
+
+OQS_API OQS_STATUS OQS_SIG_STFL_lms_sigs_total(unsigned long long *total, const OQS_SIG_STFL_SECRET_KEY *secret_key) {
+
+	uint8_t *priv_key = NULL;
+	oqs_lms_key_data *oqs_key_data = NULL;
+	struct hss_working_key *working_key = NULL;
+
+	if (total == NULL  || secret_key == NULL) {
+		return OQS_ERROR;
+	}
+
+	oqs_key_data = secret_key->secret_key_data;
+	if (!oqs_key_data) {
+		return OQS_ERROR;
+	}
+
+	priv_key = oqs_key_data->sec_key;
+	if (!priv_key) {
+		return OQS_ERROR;
+	}
+
+	working_key = hss_load_private_key(NULL, priv_key,
+	                                   0,
+	                                   NULL,
+	                                   0,
+	                                   0);
+	if (!working_key) {
+		return OQS_ERROR;
+	}
+
+	*total = (unsigned long long)working_key->max_count;
+	hss_free_working_key(working_key);
+	return OQS_SUCCESS;
+}
+
+/* LMS wrapper functions use internal OIDs to
+ * identify the parameter set to be used
+ */
+
+bool LMS_randombytes(void *buffer, size_t length) {
+
+	OQS_randombytes((uint8_t *)buffer, length);
+	return true;
+}
+
+#ifndef OQS_ALLOW_LMS_KEY_AND_SIG_GEN
+int oqs_sig_stfl_lms_keypair(UNUSED uint8_t *pk, UNUSED OQS_SIG_STFL_SECRET_KEY *sk, UNUSED const uint32_t oid) {
+	return -1;
+}
+#else
+int oqs_sig_stfl_lms_keypair(uint8_t *pk, OQS_SIG_STFL_SECRET_KEY *sk, const uint32_t oid) {
+
+	int ret = -1;
+	bool b_ret;
+	int parse_err = 0;
+
+	size_t len_public_key = 60;
+	oqs_lms_key_data *oqs_key_data = NULL;
+
+	if (!pk || !sk || !oid) {
+		return -1;
+	}
+
+	if (sk->secret_key_data) {
+		//this means a key pair has already been recreated
+		//TODO log error.
+		return -1;
+	}
+
+	oqs_key_data = malloc(sizeof(oqs_lms_key_data));
+	if (oqs_key_data == NULL) {
+		return -1;
+	}
+
+	memset(oqs_key_data, 0, sizeof(oqs_lms_key_data));
+	if (sk->length_secret_key == 0) {
+		OQS_MEM_insecure_free(oqs_key_data);
+		oqs_key_data = NULL;
+		return -1;
+	}
+
+	oqs_key_data->levels = 1;
+	oqs_key_data->len_sec_key = sk->length_secret_key;
+	oqs_key_data->sec_key = (uint8_t *)malloc(sk->length_secret_key * sizeof(uint8_t));
+	if (oqs_key_data->sec_key == NULL) {
+		OQS_MEM_insecure_free(oqs_key_data);
+		oqs_key_data = NULL;
+		return -1;
+	}
+
+	memset(oqs_key_data->sec_key, 0, sk->length_secret_key);
```

Comment:
Instead of `malloc` and then `memset`, it might be more efficient to use `calloc`.